subsys/bluetooth/host/conn.c

```diff
@@ -1088,7 +1088,15 @@ void bt_conn_security_changed(struct bt_conn *conn, enum bt_security_err err)
 	}
 #if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
 	if (!err && conn->sec_level >= BT_SECURITY_L2) {
-		bt_keys_update_usage(conn->id, bt_conn_get_dst(conn));
+		if (conn->type == BT_CONN_TYPE_LE) {
+			bt_keys_update_usage(conn->id, bt_conn_get_dst(conn));
+		}
+
+#if defined(CONFIG_BT_BREDR)
+		if (conn->type == BT_CONN_TYPE_BR) {
```

Comment:
Could this be converted to `if (IS_ENABLED(CONFIG_BT_BREDR) && conn->type == BT_CONN_TYPE_BR)`?

Reply:
> 
> 
> Could this be converted to `if (IS_ENABLED(CONFIG_BT_BREDR) && conn->type == BT_CONN_TYPE_BR)`?
This is not possible because the conn->br field is controlled by the macro

Reply:
> 
> 
> Could this be converted to `if (IS_ENABLED(CONFIG_BT_BREDR) && conn->type == BT_CONN_TYPE_BR)`?

I saw that you accepted the RP, but the test shows that conn->br requires the CONFIG_BT_BREDR macro. If you use IS_ENABLED(CONFIG_BT_BREDR), the compilation will be wrong. Do you need to roll back to the previous commit?

Reply:
GitHub will prevent merging unless the build passes, so you need to fix that. I didn't realise you made some change that broke things.